test/variables/thunks/thunk-cond.chpl

```diff
@@ -0,0 +1,21 @@
+class C {
+  var x: int;
+
+  proc init(x: int) {
+    this.x = x;
+    init this;
+    writeln("C.init");
+  }
+}
+
+proc executeIfTrue(cond: bool, in thunk: _thunkRecord) {
+  var temp: thunkToReturnType(thunk.type);
+  writeln("In execute if true, with value: ", cond);
+  if cond {
+    temp = __primitive("force thunk", thunk);
+  }
+  return temp;
+}
+
+writeln(executeIfTrue(true, __primitive("create thunk", new C?(42))));
+writeln(executeIfTrue(false, __primitive("create thunk", new C?(42))));
```

Comment:
Yep, `execIfTrue` is a generic function; if it were instantiated with a thunk that produces a non-nilable type, this line would fail to compile:

```Chapel
  var temp: thunkToReturnType(thunk.type);
```

Since it would be

```
  var temp: owned C;
```

Which doesn't have a viable default-initialization.